kernel/src/arch/x86_64/exokernel/mkdir.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2021 University of Colorado. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use abomonation::{decode, encode, Abomonation};
+use core2::io::Result as IOResult;
+use core2::io::Write;
+use log::{debug, warn};
+
+use rpc::rpc::*;
+use rpc::RPCClient;
+
+use crate::arch::exokernel::fio::*;
+use crate::cnrfs;
+
+#[derive(Debug)]
+pub struct MkDirReq {
+    pub modes: u64,
+}
+unsafe_abomonate!(MkDirReq: modes);
+
+pub fn rpc_mkdir<T: RPCClient>(
+    rpc_client: &mut T,
+    pid: usize,
+    pathname: &[u8],
+    modes: u64,
+) -> Result<(u64, u64), RPCError> {
+    debug!("MkDir({:?})", pathname);
+
+    // Construct request data
+    let req = MkDirReq { modes: modes };
+    let mut req_data = [0u8; core::mem::size_of::<MkDirReq>()];
+    unsafe { encode(&req, &mut (&mut req_data).as_mut()) }.unwrap();
+
+    // Create result buffer
+    let mut res_data = [0u8; core::mem::size_of::<FIORes>()];
```

Comment:
maybe we can template this patterns of constructing the message (happens in many functions)?

Reply:
By templating, do you mean extracting the message creation into an inline function that takes a generic type and creates it? Or something else?

Reply:
I was thinking maybe you can create a macro that you can use throughout the various RPC handler functions?